src/main/java/de/tum/cit/aet/artemis/exercise/web/ParticipationResource.java

```diff
@@ -826,11 +831,22 @@ private MappingJacksonValue participationForQuizExercise(QuizExercise quizExerci
             quizExercise.setQuizBatches(quizBatch.stream().collect(Collectors.toSet()));
             quizExercise.filterForStudentsDuringQuiz();
             StudentParticipation participation = participationForQuizWithResult(quizExercise, user.getLogin(), quizBatch.get());
-            // set view
-            var view = quizExercise.viewForStudentsInQuizExercise(quizBatch.get());
-            MappingJacksonValue value = new MappingJacksonValue(participation);
-            value.setSerializationView(view);
-            return value;
+
+            // TODO: Duplicate
+            Object responseDTO = null;
+            if (participation != null) {
+                if (quizExercise.isQuizEnded()) {
+                    responseDTO = StudentQuizParticipationWithSolutionsDTO.of(participation);
+                }
+                else if (quizBatch.get().isStarted()) {
+                    responseDTO = StudentQuizParticipationWithQuestionsDTO.of(participation);
+                }
+                else {
+                    responseDTO = StudentQuizParticipationWithoutQuestionsDTO.of(participation);
+                }
+            }
+
+            return responseDTO != null ? new MappingJacksonValue(responseDTO) : null;
```

Comment:
These null checks is either redundant or you should probably throw a proper exception